core/version.h

```diff
@@ -3,9 +3,20 @@
  * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
```

Comment:
given we have `.in` file now, does this file still have to be checked-in?

Reply:
thanks for reminding.

Reply:
@no1wudi We are considering generating the *version.h* file from *version.h.in* during the CMake configuration process and removing version.h from the source code tree. [The Nuttx CI](https://github.com/bytecodealliance/wasm-micro-runtime/actions/runs/12705810342/job/35417551343?pr=3933) is failing because it uses make instead of CMake and cannot create the version.h file. Is there a method to still generate version.h using make?

Reply:
@loganek Several embedded platforms rely on Makefile instead of CMake and are unable to utilize `configure_file()` or other alternatives(like sed) to generate *version.h* from *version.h.in*. Therefore, it is advisable to continue maintaining version.h within the source code tree.

Reply:
ok understood. I wonder though if we can ensure in CI that those two files (`version.h.in` and `version.h`) stay in sync?

Reply:
something like https://github.com/bytecodealliance/wasm-micro-runtime/pull/3933/files#diff-f275178c626161329dae1b5a6505c032a697ec141e65cdbca0df7b45112f63d3 ?

Reply:
that's perfect, thanks!

Reply:
@loganek  If there are no further comments, shall we go ahead and merge the pull request?